tests/test_optimizer.py

```diff
@@ -374,12 +374,12 @@ def run_trainer(batch_size: int, batch_time: float, client_mode: bool):
         peer.join()
 
     assert isinstance(optimizer, Optimizer)
-    assert optimizer.local_epoch == optimizer.tracker.global_epoch == total_epochs
+    assert optimizer.local_epoch == optimizer._tracker.global_epoch == total_epochs
```

Comment:
I believe that using private fields in tests in fine

Reply:
I agree.

Reply:
These fields are *not* private and are meant to be exposed to the user.

Reply:
Perhaps it is indeed a significant change given current usage: however, I must point out that Optimizer was an experimental class up to this PR and didn't appear in any of the releases, so its API was subject to change at any moment :)

Overall, I agree with not changing these fields to private right away, but we should do it in the next major release due to the reasons specified in the PR message. By that time, we should have a pretty good understanding of what internals of tracker/averager need to be made available by Optimizer